crates/iota-json-rpc/src/governance_api.rs

```diff
@@ -638,8 +644,34 @@ fn pending_validator_exchange_rate(
         })
         .collect::<Vec<(IotaAddress, ObjectID, ObjectID, u64, bool)>>();
 
+    // Try to find for validator candidate and get its exchange rates
+    for (_, df) in state.get_dynamic_fields(
+        system_state_summary.validator_candidates_id,
+        None,
+        system_state_summary.validator_candidates_size as usize,
+    )? {
+        let raw_address =
+            String::deserialize(df.name.value).map_err(|err| IotaError::from(err.to_string()))?;
+        let validator_address = IotaAddress::from_str(&raw_address)?;
```

Comment:
This error here turns out to be `iota_json_rpc::error::Error::Internal` due to conversion from the original `anyhow` error in the calling method. `StateReadError` seems proper here as well to give more context.

Reply:
resolving in favor of: https://github.com/iotaledger/iota/pull/4313#discussion_r1865396883